src/components/NcHeaderMenu/NcHeaderMenu.vue

```diff
@@ -0,0 +1,359 @@
+			:aria-expanded="opened.toString()"
+			@click.prevent="toggleMenu">
+			<!-- @slot Icon trigger slot. Make sure the svg path
+				is at least 16px. Usually mdi icon works at 20px -->
+			<slot name="trigger" />
+		</a>
+
+		<!-- Visual triangle -->
+		<div v-show="opened" class="header-menu__carret" />
+
+		<!-- Menu opened content -->
+		<div v-show="opened"
+			:id="`header-menu-${id}`"
+			class="header-menu__wrapper"
+			role="menu">
+			<div ref="content" class="header-menu__content">
+				<!-- @slot Main content -->
+				<slot />
+			</div>
+		</div>
+	</div>
+</template>
+
+<script>
+import { directive as ClickOutside } from 'v-click-outside'
+import { createFocusTrap } from 'focus-trap'
+
+import excludeClickOutsideClasses from '../../mixins/excludeClickOutsideClasses/index.js'
+import { getTrapStack } from '../../utils/focusTrap.js'
+
+export default {
+	name: 'NcHeaderMenu',
+
+	directives: {
+		ClickOutside,
+	},
+
+	mixins: [
+		excludeClickOutsideClasses,
+	],
+
+	props: {
+		/**
+		 * Unique id for this menu
+		 */
+		id: {
+			type: String,
+			required: true,
+		},
+
+		/**
+		 * aria-label attribute of the menu open button
+		 */
+		ariaLabel: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * Current menu open state
+		 */
+		open: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	emits: [
+		'close',
+		'open',
+		'update:open',
+		'cancel',
+	],
+
+	data() {
+		return {
+			focusTrap: null,
+			opened: this.open,
+			shortcutsDisabled: window.OCP?.Accessibility?.disableKeyboardShortcuts?.(),
+
+			clickOutsideConfig: {
+				handler: this.closeMenu,
+				middleware: this.clickOutsideMiddleware,
+			},
+		}
+	},
+
+	watch: {
+		open(newVal) {
+			this.opened = newVal
+			// Wait for component to finish rendering
+			this.$nextTick(() => {
+				// Trigger events
+				this.opened ? this.openMenu() : this.closeMenu()
+			})
+		},
+	},
+
+	mounted() {
+		document.addEventListener('keydown', this.onKeyDown)
+	},
+	beforeDestroy() {
+		document.removeEventListener('keydown', this.onKeyDown)
+	},
+
+	methods: {
+		/**
+		 * Toggle the current menu open state
+		 */
+		toggleMenu() {
+			// Toggling current state
+			if (!this.opened) {
+				this.openMenu()
+			} else {
+				this.closeMenu()
+			}
+		},
+
+		/**
+		 * Close the current menu
+		 */
+		closeMenu() {
+			if (!this.opened) {
+				return
+			}
+
+			this.opened = false
+			this.$emit('close')
+			this.$emit('update:open', false)
+
+			// Kill focus trap
+			this.clearFocusTrap()
+		},
+
+		/**
+		 * Open the current menu
+		 */
+		openMenu() {
+			if (this.opened) {
+				return
+			}
+
+			this.opened = true
+			this.$emit('open')
+			this.$emit('update:open', true)
+
+			/**
+			 * Triggered after the tooltip was visually displayed.
+			 *
+			 * This is different from the 'show' and 'apply-show' which
+			 * run earlier than this where there is no guarantee that the
+			 * tooltip is already visible and in the DOM.
+			 */
+			this.$nextTick(() => {
```

Comment:
There is already one in `useFocusTrap`, is this one needed ?

Reply:
Done